projects/client/RabbitMQ.Client/src/client/impl/AsyncConsumerWorkService.cs

```diff
@@ -1,69 +1,55 @@
-﻿using System;
+﻿using RabbitMQ.Client.Impl;
+
 using System.Collections.Concurrent;
 using System.Threading;
 using System.Threading.Tasks;
-using RabbitMQ.Client.Impl;
 
 namespace RabbitMQ.Client
 {
     internal sealed class AsyncConsumerWorkService : ConsumerWorkService
     {
-        readonly ConcurrentDictionary<IModel, WorkPool> workPools = new ConcurrentDictionary<IModel, WorkPool>();
+        private readonly ConcurrentDictionary<IModel, WorkPool> workPools = new ConcurrentDictionary<IModel, WorkPool>();
 
-        public void Schedule<TWork>(ModelBase model, TWork work)
-            where TWork : Work
+        public void Schedule<TWork>(ModelBase model, TWork work) where TWork : Work
         {
-            // two step approach is taken, as TryGetValue does not aquire locks
-            // if this fails, GetOrAdd is called, which takes a lock
-
-            WorkPool workPool;
-            if (workPools.TryGetValue(model, out workPool) == false)
-            {
-                var newWorkPool = new WorkPool(model);
-                workPool = workPools.GetOrAdd(model, newWorkPool);
-
-                // start if it's only the workpool that has been just created
-                if (newWorkPool == workPool)
-                {
-                    newWorkPool.Start();
-                }
-            }
+            workPools.GetOrAdd(model, StartNewWorkPool).Enqueue(work);
```

Comment:
Not at all performance sensitive code, so it's much more readable like this, but will revert if needed.

Reply:
This seems reasonable.